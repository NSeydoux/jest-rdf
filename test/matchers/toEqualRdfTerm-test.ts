import {blankNode, namedNode, variable} from "rdf-data-model";
import "../../index";

describe('#toEqualRdfTerm', () => {
  it('should succeed for equal named nodes', () => {
    return expect(namedNode('a')).toEqualRdfTerm(namedNode('a'));
  });

  it('should not succeed for non-equal named nodes', () => {
    return expect(namedNode('a')).not.toEqualRdfTerm(namedNode('b'));
  });

  it('should not succeed for a named node and a variable', () => {
    return expect(namedNode('a')).not.toEqualRdfTerm(variable('a'));
  });

  it('should not fail for equal named nodes', () => {
    return expect(() => expect(namedNode('a')).not.toEqualRdfTerm(namedNode('a'))).toThrowErrorMatchingSnapshot();
  });

  it('should fail for non-equal named nodes', () => {
    return expect(() => expect(namedNode('a')).toEqualRdfTerm(namedNode('b'))).toThrowErrorMatchingSnapshot();
  });

  it('should fail for a named node and a variable', () => {
    return expect(() => expect(namedNode('a')).toEqualRdfTerm(variable('a'))).toThrowErrorMatchingSnapshot();
  });

  it('should succeed for equal blank nodes', () => {
    return expect(blankNode('a')).toEqualRdfTerm(blankNode('a'));
  });

  it('should succeed for non-equal blank nodes', () => {
    return expect(blankNode('a')).toEqualRdfTerm(blankNode('b'));
  });

  it('should not succeed for a blank node and a variable', () => {
    return expect(blankNode('a')).not.toEqualRdfTerm(variable('a'));
  });

  it('should not fail for equal blank nodes', () => {
    return expect(() => expect(blankNode('a')).not.toEqualRdfTerm(blankNode('a'))).toThrowErrorMatchingSnapshot();
  });

  it('should not fail for non-equal blank nodes', () => {
    return expect(() => expect(blankNode('a')).not.toEqualRdfTerm(blankNode('b'))).toThrowErrorMatchingSnapshot();
  });

  it('should fail for a blank node and a variable', () => {
    return expect(() => expect(blankNode('a')).toEqualRdfTerm(variable('a'))).toThrowErrorMatchingSnapshot();
  });
});
