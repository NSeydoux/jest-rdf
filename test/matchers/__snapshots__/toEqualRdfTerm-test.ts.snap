// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toEqualRdfTerm should fail for a blank node and a quads 1`] = `"expected _:a and <<s p o g>> to be equal"`;

exports[`#toEqualRdfTerm should fail for a blank node and a variable 1`] = `"expected _:a and ?a to be equal"`;

exports[`#toEqualRdfTerm should fail for a named node and a variable 1`] = `"expected a and ?a to be equal"`;

exports[`#toEqualRdfTerm should fail for non-equal named nodes 1`] = `"expected a and b to be equal"`;

exports[`#toEqualRdfTerm should fail for non-equal quads 1`] = `"expected <<s p o g>> and <<s1 p o g>> to be equal"`;

exports[`#toEqualRdfTerm should not fail for equal blank nodes 1`] = `"expected _:a and _:a not to be equal"`;

exports[`#toEqualRdfTerm should not fail for equal named nodes 1`] = `"expected a and a not to be equal"`;

exports[`#toEqualRdfTerm should not fail for equal quads 1`] = `"expected <<s p o g>> and <<s p o g>> not to be equal"`;

exports[`#toEqualRdfTerm should not fail for non-equal blank nodes 1`] = `"expected _:a and _:b not to be equal"`;
